### PR TITLE
feat: adjust browser-level cookie API

### PR DESCRIPTION
**Pull request by OrKoN**
- move new API's tests to a separate spec.
- use a different type for setting cookies as not all attributes of Cookie are applicable.
- fix various tests requiring secure origins.
- fix the issue with always using default context to set cookies.

### CODE REVIEW

File: packages/puppeteer-core/src/bidi/Page.ts [hunk 1]
```diff
@@ -929,14 +929,25 @@ function testUrlMatchCookie(cookie: Cookie, url: URL): boolean {
   return testUrlMatchCookiePath(cookie, normalizedUrl);
 }
 
-export function bidiToPuppeteerCookie(bidiCookie: Bidi.Network.Cookie): Cookie {
+export function bidiToPuppeteerCookie(
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/bidi/Page.ts:932]: Non blocking: Should we move these out of Page an into separate file to make it clear that they are utility now that we are exporting them?

File: packages/puppeteer-core/src/bidi/Page.ts [hunk 2]
```diff
@@ -1028,10 +1039,15 @@ export function convertCookiesExpiryCdpToBiDi(
...
-  partitionKey: ChromeCookiePartitionKey | string | undefined,
+  partitionKey: CookiePartitionKey | string | undefined,
 ): string | undefined {
   if (partitionKey === undefined || typeof partitionKey === 'string') {
     return partitionKey;
   }
-  return partitionKey.topLevelSite;
+  if (partitionKey.hasCrossSiteAncestor) {
+    throw new UnsupportedOperation(
+      'WebDriver BiDi does not support goog:hasCrossSiteAncestor yet.',
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/bidi/Page.ts]: I think we don't want to mention the prefix
```suggestion
      'WebDriver BiDi does not support `hasCrossSiteAncestor` yet.',
```

File: test/src/browsercontext-cookies.spec.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc.
```
Comment (Lightning00Blade) [inline test/src/browsercontext-cookies.spec.ts]:
```suggestion
 * Copyright 2024 Google Inc.
```

File: packages/puppeteer-core/src/bidi/Page.ts (hunk 1 above)
Comment (OrKoN) [reply packages/puppeteer-core/src/bidi/Page.ts:932]: I think we can move them completely to BrowserContext once we remove the page-level cookies API.

**Review by Lightning00Blade — APPROVED** [review]

**Pull request merged**